File: problems/0001 - 0400/0069/0069.cpp
class Solution {
public:
    int mySqrt(int x) {
        unsigned int start = 1;
        unsigned int end = x;
        while (start <= end) {
            unsigned int mid = start + (end - start) / 2;
            unsigned long long mid2 = (unsigned long long)mid * mid;
            if (mid2 == x) {
                return mid;
            }
            else if (mid2 > x) {
                end = mid - 1;
            }
            else {
                start = mid + 1;
            }
        }
        return end;
    }
};

File: problems/0001 - 0400/0069/0069.py
class Solution:
    def mySqrt(self, x: int) -> int:
        start = 1
        end = x
        while start <= end:
            mid = start + (end - start) // 2
            mid2 = mid * mid
            if mid2 == x:
                return mid
            elif mid2 < x:
                start = mid + 1
            else:
                end = mid - 1
        return end
